src/python/pants/backend/go/target_types.py

```diff
@@ -69,20 +111,23 @@ class GoModDependenciesField(Dependencies):
     alias = "_dependencies"
 
 
-class GoModTarget(Target):
+class GoModTarget(TargetGenerator):
     alias = "go_mod"
-    core_fields = (
-        *COMMON_TARGET_FIELDS,
-        GoModDependenciesField,
-        GoModSourcesField,
-    )
     help = (
         "A first-party Go module (corresponding to a `go.mod` file).\n\n"
         "Generates `go_third_party_package` targets based on the `require` directives in your "
         "`go.mod`.\n\n"
         "If you have third-party packages, make sure you have an up-to-date `go.sum`. Run "
         "`go mod tidy` directly to update your `go.mod` and `go.sum`."
     )
+    generated_target_cls = GoThirdPartyPackageTarget
+    core_fields = (
+        *COMMON_TARGET_FIELDS,
+        GoModDependenciesField,
```

Comment:
prob add comment why not moved

Reply:
wait we dont use deps on py reqs generators? maybe we don't need it here?

Reply:
Yea: see the comment on the `GoModDependenciesField` above. I don't know how to deprecate a field for removal (as opposed to for renaming), so I punted on doing anything about that comment.